.github/workflows/ngsa-pr.yaml

```diff
@@ -27,16 +27,45 @@ jobs:
         registry: ghcr.io
         username: ${{ secrets.GHCR_ID }}
         password: ${{ secrets.GHCR_PAT }}
-
+    - name: Set Secrets
+      run: |
+            echo -n ${{ secrets.NGSA_PRE_COSMOS_RO_KEY}} >| secrets/CosmosKey
+            echo -n ${{ secrets.NGSA_PRE_COSMOS_URL }} >| secrets/CosmosUrl
     - name: Docker Build
       run: |
-        docker build . -t image
-        
+        docker build . --progress auto -t image
+
+    - name: Docker Run and Test
+      run: |
+            #docker run --rm image --version
+            #docker network create runandtestnet 
+            #docker run --rm --net=runandtestnet --network-alias=ngsacsapi --name ngsacs -p 8080:8080 image
+            
+            docker run --rm -d --name ngsacs -p 8080:8080 image
+
+            echo "Waiting for web server to start ..."
```

Comment:
why is the sleep hard-coded? In the previous version we used a loop

Reply:
Utilized While loop logic to wait 10 secs, then wait for 1 sec up to 30 secs then time out